### PR TITLE
wr-314: Add sorting to the filters, redesign the Follow button, remove the dropdown menu for articles by other users

### PR DESCRIPTION
**Pull request by AnastasiaR2**
- [x] The dropdown menu icon should be displayed only for articles authored by the user.
- [x] Authors in the Author filter’s dropdown list should be sorted in alphabetical order.
- [x] Genres in the Genre filter’s dropdown list should be sorted alphabetically.
- [x] The user’s avatar should not change its size during profile editing.
- [x] Redesign the Follow button and add some padding.

image
image
image

### CODE REVIEW

File: backend/src/packages/users/user.repository.ts [hunk 1]
```diff
@@ -163,12 +164,10 @@ class UserRepository implements IRepository {
...
 
   public async getAllAuthors(): Promise<UserDetailsModel[] | null> {
-    const authors = await UserDetailsModel.query().distinctOn('userId')
-      .whereRaw(`
-      EXISTS(SELECT 1
-      FROM articles,user_details
-      WHERE articles.user_id = user_details.user_id)
-    `);
+    const authors = await UserDetailsModel.query()
+      .from(getAuthorsSubquery())
```
Comment (andanf-e) [inline backend/src/packages/users/user.repository.ts]: Let's keep this query in repository - as from helper it is used only single time here

File: backend/src/packages/users/user.repository.ts [hunk 2]
```diff
@@ -163,12 +163,24 @@ class UserRepository implements IRepository {
...
-      EXISTS(SELECT 1
-      FROM articles,user_details
-      WHERE articles.user_id = user_details.user_id)
-    `);
+    const authors = await UserDetailsModel.query()
+      .from((builder) => {
+        void builder
+          .distinctOn('userId')
+          .select('userId', 'firstName', 'lastName')
+          .from('user_details')
```
Comment (andanf-e) [inline backend/src/packages/users/user.repository.ts:171]: Are you sure we need to have separated .from statement as it's already in UserDetailsModel

Comment (AnastasiaR2) [reply backend/src/packages/users/user.repository.ts:171]: If I remove separated .from and leave the code like so
```
const authors = await UserDetailsModel.query()
      .distinctOn('userId')
      .select('userId', 'firstName', 'lastName')
      .from('user_details')
      .whereRaw(
        `
        EXISTS(SELECT 1
        FROM articles,user_details
        WHERE articles.user_id = user_details.user_id)
      `,
      )
      .orderBy('userId')
      .as('subquery')
      .orderBy('firstName')
      .orderBy('lastName');
```
the sorting doesn't work. Did I miss something?

Comment (andanf-e) [reply backend/src/packages/users/user.repository.ts:171]: Not sure that aliasing is needed, and I believe at first you should order by firstname\lastname,, and only after that with userId

Comment (andanf-e) [reply backend/src/packages/users/user.repository.ts:171]: And let's have sorting on authors with `orderBy('CONCAT(first_name, ' ', last_name)')` if it's possible

Comment (AnastasiaR2) [reply backend/src/packages/users/user.repository.ts:171]: As I have checked, the expressions specified in the DISTINCT ON clause must also appear in the ORDER BY clause in the same order.
So, it's not possible to order by firstname\lastname at first. That's why, I used a subquery here.

Comment (andanf-e) [reply backend/src/packages/users/user.repository.ts:171]: Okay

**Review by andanf-e — DISMISSED** [review]

File: backend/src/packages/users/user.repository.ts [hunk 3]
```diff
@@ -163,12 +163,24 @@ class UserRepository implements IRepository {
...
+          .whereRaw(
+            `
+        EXISTS(SELECT 1
+        FROM articles
+        WHERE articles.user_id = user_details.user_id)
```
Comment (AnastasiaR2) [inline backend/src/packages/users/user.repository.ts:176]: I noticed that all users are displayed as authors, even if no articles are written by the user. Updated query to resolve this.

**Review by andanf-e — APPROVED** [review]

**Pull request merged**